File: solutions/c/easy/pointer-in-c/solution-approach1.c
// ──────────────────────────────────────────────────
// Problem     Pointers in C
// Difficulty  Easy
// Subdomain   Introduction
// Platform    HackerRank
// Language    c
// Status      Accepted
// Submitted   2026-06-10, 09:47 p.m.
// ──────────────────────────────────────────────────

#include <stdio.h>

void update(int *a,int *b) {
    // Complete this function  
      int sum = *a + *b;
    int diff = *a - *b;

    if(diff < 0)
    {
        diff = -diff;
    }

    *a = sum;
    *b = diff;  
}

int main() {
    int a, b;
    int *pa = &a, *pb = &b;
    
    scanf("%d %d", &a, &b);
    update(pa, pb);
    printf("%d\n%d", a, b);

    return 0;
}

File: solutions/python3/easy/list-comprehensions/solution-approach1.py
# ──────────────────────────────────────────────────
# Problem     List Comprehensions
# Difficulty  Easy
# Subdomain   Basic Data Types
# Platform    HackerRank
# Language    python3
# Status      Accepted
# Submitted   2026-06-25, 05:37 p.m.
# ──────────────────────────────────────────────────

if __name__ == '__main__':
    x = int(input())
    y = int(input())
    z = int(input())
    n = int(input())
    # Single list comprehension replacing nested loops
    coordinates = [[i, j, k] for i in range(x + 1) for j in range(y + 1) for k in range(z + 1) if i + j + k != n]
    
    print(coordinates)
